app/main.py

```diff
@@ -1 +1,45 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills: list[str] = []
+
+    def learn_skill(self, skill: str) -> None:
+        if skill not in self.skills:
+            self.skills.append(skill)
```

Comment:
you can not use `if` in `learn_skill` method, if you have created list in the `__init__ `method, because you already have this list

Reply:
Method **__init__** creates an empty list for storing skills.
The **learn_skill** method adds a new skill, but first checks if the list already contains this skill to avoid duplicates.
So, in my opinion, the verification is important.